### PR TITLE
Add riscv64 cross-build infra

### PR DESCRIPTION
**Pull request by am11**
<del>This PR contains a commit from #10153. I'll rebase once the other PR is merged.</del>

RISC-V currently has 32, 64 and 128 bits configurations (128 is yet to be ratified), this PR is only adding RISCV64 support with `G`;  general purpose, extension (https://en.wikipedia.org/wiki/RISC-V#ISA_base_and_extensions).

The version of HP libunwind (1.6.2) sources in runtime repo has the support for `riscv64`, but Debian package is stuck at 1.3.2 for quite some time https://tracker.debian.org/pkg/libunwind. Therefore, I have excluded it from restore (we don't need `libunwind8-dev` package for our current matrix and .NET version, but theoretically SourceBuild can use this script to cross compile with "use system libunwind" switch turned on).

LLDB does not support System-V ABI for RISC-V, which we use in diagnostics repo for SOS plugin, so I have dropped it as well.

All the other dependencies that we need are available in sid's default repository (`unstable`).

Once the docker image is built, it will help folks who are interested in porting runtime to RISC-V without waiting for baremetal.

cc @jkotas, @janvorli

### CODE REVIEW

File: eng/common/cross/toolchain.cmake [hunk 1]
```diff
@@ -80,8 +80,11 @@ elseif(TARGET_ARCH_NAME STREQUAL "x64")
   elseif(ILLUMOS)
     set(TOOLCHAIN "x86_64-illumos")
   endif()
+elseif(TARGET_ARCH_NAME STREQUAL "riscv64")
```
Comment (akoeplinger) [inline eng/common/cross/toolchain.cmake:83]: nit: I'd move this to be below s390x so it matches the order of the arch names in the FATAL_ERROR below

File: eng/common/cross/toolchain.cmake [hunk 2]
```diff
@@ -80,8 +80,11 @@ elseif(TARGET_ARCH_NAME STREQUAL "x64")
...
+  set(CMAKE_SYSTEM_PROCESSOR riscv64)
+  set(TOOLCHAIN "riscv64-linux-gnu")
 else()
-  message(FATAL_ERROR "Arch is ${TARGET_ARCH_NAME}. Only armel, arm, armv6, arm64, ppc64le, s390x and x86 are supported!")
+  message(FATAL_ERROR "Arch is ${TARGET_ARCH_NAME}. Only armel, arm, armv6, arm64, ppc64le, s390x, riscv64 and x86 are supported!")
```
Comment (akoeplinger) [inline eng/common/cross/toolchain.cmake:87]: just saw that we're missing `x64` in the list here, would you mind adding it? :)

Comment (am11) [reply eng/common/cross/toolchain.cmake]: I have alphabetically sorted them in both, toolchain and build-rootfs.

**Comment by akoeplinger** [comment]
this need a rebase

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**